cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -47,68 +61,107 @@ public ApplicationsOperations(DefaultCloudFoundryOperations cloudFoundryOperatio
     }
 
     /**
-     * Prepares a request for fetching applications data from the cloud foundry instance.
-     * The resulting mono will not perform any logging by default.
+     * Prepares a request for fetching applications data from the cloud foundry
+     * instance. The resulting mono will not perform any logging by default.
      *
-     * @return mono object of all applications as map of the application names as key and the ApplicationBeans as value
+     * @return mono object of all applications as map of the application names as
+     *         key and the ApplicationBeans as value
      */
     public Mono<Map<String, ApplicationBean>> getAll() {
+
         return this.cloudFoundryOperations
-                .applications()
-                .list()
-                .flatMap(this::getApplicationManifest)
-                .collectMap(ApplicationManifest::getName, ApplicationBean::new);
+            .applications()
+            .list()
+            .flatMap(applicationSummary -> Flux.zip(
+                getApplicationManifest(applicationSummary),
+                getMetadata(applicationSummary))
+                .collectMap(tuple -> tuple.getT1().getName(),
+                    tuple -> new ApplicationBean(tuple.getT1(), tuple.getT2())))
+            .reduce(new HashMap<String, ApplicationBean>(), (newMap, currentMap) -> {
+                newMap.putAll(currentMap);
```

Comment:
In my opinion the code is not so easy to understand. Some comments explaining what is going on here would be helpful.

Reply:
Quite tricky to implement this. :)
One solution that might be a little easier to understand:

```suggestion
           this.cloudFoundryOperations
                .applications()
                .list()
                .flatMap(applicationSummary -> Flux.zip(this.getApplicationManifest(applicationSummary), this.getMetadata(applicationSummary)))
                // T1 == applicationManifest, T2 == metadata
                .collectMap(tuple -> tuple.getT1().getName(), tuple -> new ApplicationBean(tuple.getT1(), tuple.getT2()));
```